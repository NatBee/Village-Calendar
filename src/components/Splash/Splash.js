import React, { Component } from 'react';
import { quotes } from '../../assets/quotes';
import Register from '../../containers/Register/Register';
import './Splash.css';
import { Link } from 'react-router-dom';
import { logInUser, setToken } from '../../actions/index';
import { connect } from 'react-redux';
import firebase from 'firebase/app';
import 'firebase/auth';
import { config } from '../../helper/apiKey';
import asyncLoader from 'react-async-loader';

firebase.initializeApp(config);

class Splash extends Component {
  constructor(props) {
    super(props);
    this.state = {
      quote: '',
      author: ''
    }
  }

  componentDidMount() {
    this.randomNumber();
  }
  
  randomNumber = () => {
    const randomIndex = Math.floor(Math.random() * Math.floor(9));
    this.quoteDisplay(randomIndex);
  }

  quoteDisplay = (index) => {
    const quoteObj = quotes[index];
    const quote = quoteObj.quote;
    const author = quoteObj.author;

    this.setState ({ quote, author })
  }

  logIn = async () => {
    const provider = new firebase.auth.GoogleAuthProvider();
    provider.addScope('https://www.googleapis.com/auth/calendar');
    const auth = firebase.auth()
    const authentication = await auth.signInWithPopup(provider)
    this.props.logInUser(authentication);

    const token = authentication.credential.accessToken
    localStorage.setItem('ouath2-access-token', JSON.stringify(token))
    this.props.setToken(token);

    this.props.history.push('/calendar');
  } 

  buttonDisplay = () => {
    if(this.props.user.user === undefined) {
      return(
        <button className="login" onClick={this.logIn}>Log In</button>
      )
    }
  }

  renderContainer = () => {
    const route = this.props.location.pathname;
    if(route === "/") {
      return (
        <div>
          {this.buttonDisplay()}
          <Link to="/register">Create Your Village</Link>  
        </div>
      )
    } else if(route === "/register") {
      return <Register />
    }
  }

  render() {
    return (
      <div>
        <h3>{ this.state.quote }</h3>
        <h4>{ this.state.author }</h4>
        { this.renderContainer()}
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  user: store.user,
  token: store.token
})

export const mapDispatchToProps = (dispatch) => ({
  logInUser: (user) => dispatch(logInUser(user)),
  setToken: (token) => dispatch(setToken(token))
})

export default connect(mapStateToProps, mapDispatchToProps)(Splash);