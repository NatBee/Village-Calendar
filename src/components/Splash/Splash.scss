.calendar {
  padding-bottom: 120px;
  text-align: -webkit-center;
  font-family: 'Exo 2', sans-serif;
} 

.intro {
  font-family: 'Exo', sans-serif;
  width: 515px;
  padding: 10px 0 20px;
  font-size: .75em;
}

.intro li {
  text-align: justify;
}