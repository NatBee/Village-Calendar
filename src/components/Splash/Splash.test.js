/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Splash from './Splash';
import { shallow } from 'enzyme';

describe('splash', () => {
  it('should match snapshot', () => {
    const DATE_TO_USE = new Date('2016');
    const _Date = Date;
    global.Date = jest.fn(() => DATE_TO_USE);
    global.Date.UTC = _Date.UTC;
    global.Date.parse = _Date.parse;
    global.Date.now = _Date.now;

    const mockLocation = {
      pathname: '/'
    }
    const wrapper = shallow(<Splash location={ mockLocation }/>);
    expect(wrapper).toMatchSnapshot();
  })
}) 