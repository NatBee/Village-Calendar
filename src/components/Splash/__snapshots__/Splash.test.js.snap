// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splash should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash
    location={
      Object {
        "pathname": "/",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Light is the task where many share the toil.
        </h3>,
        <h4>
          Homer
        </h4>,
        <div>
          <Link
            replace={false}
            to="/register"
          >
            Create Your Village
          </Link>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Light is the task where many share the toil.",
        },
        "ref": null,
        "rendered": "Light is the task where many share the toil.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Homer",
        },
        "ref": null,
        "rendered": "Homer",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/register"
          >
            Create Your Village
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Create Your Village",
            "replace": false,
            "to": "/register",
          },
          "ref": null,
          "rendered": "Create Your Village",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Light is the task where many share the toil.
          </h3>,
          <h4>
            Homer
          </h4>,
          <div>
            <Link
              replace={false}
              to="/register"
            >
              Create Your Village
            </Link>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Light is the task where many share the toil.",
          },
          "ref": null,
          "rendered": "Light is the task where many share the toil.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Homer",
          },
          "ref": null,
          "rendered": "Homer",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/register"
            >
              Create Your Village
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Create Your Village",
              "replace": false,
              "to": "/register",
            },
            "ref": null,
            "rendered": "Create Your Village",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
